erc721/contracts/ERC721.huff

```diff
@@ -92,7 +93,32 @@
 }
 
 #define macro SUPPORTS_INTERFACE() = takes (0) returns (0) {
-    0x00 0x00 revert
+    // grab interfaceId
+    0x04 calldataload       // [interfaceId]
+    0xe0 
+    shr
+
+    // check if erc165 interfaceId
+    dup1                    // [interfaceId, interfaceId]
+    0x01ffc9a7 eq           // [is_erc165, interfaceId]
+    is_interface jumpi
+
+    // check if erc721 interfaceId
+    dup1                    // [interfaceId, interfaceId]
+    0x80ac58cd eq           // [is_erc721, interfaceId]
+    is_interface jumpi
+
+    // check if erc721Metadata interfaceId
+    0x5b5e139f eq           // [is_erc721Metadata]
+    is_interface jumpi
```

Comment:
Can we move these interface ids into constants? ie:
```
#define constant IS_ERC165 = 0x01ffc9a7
#define constant IS_ERC721 = 0x80ac58cd
#define constant IS_ERC721_METADATA = 0x5b5e139f
```

Then use bracket notation to push constants onto the stack, ie: `[IS_ERC165] eq` for line 108.